### DISCUSSION TITLE
Recognize Classifier process error after update to AIO v4.3.4

### DISCUSSION THREAD

**Discussion by qudiqudi** [nextcloud/all-in-one#2012] [Questions] [answered]
I updated to 4.3.4 from 4.2.0 last night and had to find out that Recognize no longer wants to work properly. It reports that object recognition does not work. Though it seems that face recognition also is not working. See logs and Screenshot below.
I already re-installed the app via the GUI which did not help at all.

### Steps to reproduce
1. Use Recognize App
2. Activate object recognition in WASM mode
3. See Error in Recognize GUI and NC logs

### Expected behavior 
Object recognition works

### Actual behavior 
Error is shown and object recognition does not work

### Host OS 
Docker on Debian

#### Nextcloud AIO version 
v4.3.4

#### Current channel 
latest

#### Other valuable info 
```
[core] Fehler: RuntimeException: Classifier process error at <<closure>>

0. /var/www/html/custom_apps/recognize/lib/Classifiers/Images/ImagenetClassifier.php line 47
   OCA\Recognize\Classifiers\Classifier->classifyFiles("imagenet", [OCA\Recognize\D ... "], 600)
1. /var/www/html/custom_apps/recognize/lib/BackgroundJobs/ClassifyImagenetJob.php line 41
   OCA\Recognize\Classifiers\Images\ImagenetClassifier->classify([OCA\Recognize\D ... "])
2. /var/www/html/custom_apps/recognize/lib/BackgroundJobs/ClassifierJob.php line 70
   OCA\Recognize\BackgroundJobs\ClassifyImagenetJob->classify([OCA\Recognize\D ... "])
3. /var/www/html/custom_apps/recognize/lib/BackgroundJobs/ClassifyImagenetJob.php line 33
   OCA\Recognize\BackgroundJobs\ClassifierJob->runClassifier("imagenet", [7,174])
4. /var/www/html/lib/public/BackgroundJob/Job.php line 78
   OCA\Recognize\BackgroundJobs\ClassifyImagenetJob->run([7,174])
5. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 103
   OCP\BackgroundJob\Job->start(OC\BackgroundJob\JobList {})
6. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 93
   OCP\BackgroundJob\TimedJob->start(OC\BackgroundJob\JobList {})
7. /var/www/html/cron.php line 152
   OCP\BackgroundJob\TimedJob->execute(OC\BackgroundJob\JobList {}, OC\Log {})

at 2023-02-17T07:29:44+00:00
```

```
[recognize] Warnung: Classifier process output: sh: exec: line 0: /var/www/html/custom_apps/recognize/bin/node: not found


at 2023-02-17T07:29:44+00:00
```

<img width="925" alt="Bildschirm­foto 2023-02-17 um 08 49 35" src="https://user-images.githubusercontent.com/5947641/219583249-addc79f7-4ea6-48fa-a66e-2e74eda0633b.png">

<img width="929" alt="Bildschirm­foto 2023-02-17 um 09 19 02" src="https://user-images.githubusercontent.com/5947641/219590677-cc29bc15-faae-4233-887e-93a86537f546.png">

**Comment by szaimen** [comment]
Hi, you could check if adding `gcompat` as additional dependency makes it work by using https://github.com/nextcloud/all-in-one#how-to-add-os-packages-permanently-to-the-nextcloud-container. We removed the dependency with 4.3.0 as it breaks the app on some instances.

**Comment by qudiqudi** [comment]
I added gcompat and checked for it's presence in the nextcloud container with ´apk info´, just to be sure. On itself it did not change anything, Recognize is just not working. I then re-installed the Recognize app again, downloaded the models activated WASM-mode and it unfortunately did not change it either. Other than Recognize is constantly checking for libtensorflow like in 4.2.0
![grafik](https://user-images.githubusercontent.com/5947641/219689091-f541d89b-93a1-401c-9d51-0c53ffdddd20.png)
nothing has changed and Recognize is broken. Plus the logs now do not show anything related to Recognize, unlike before adding gcompat...

**Comment by szaimen** [comment, marked as answer]
Yes, the `stuck checking libtensorflow` issue was fixed by removing gcompat. Anyhow I dont see anythinf we can improve in AIO to fix this. It is likely an incompatibility of the libraries used in recognize with Alpine Linux that is used as base for the Nextcloud container.

**Reply by qudiqudi** [reply]
Weird though, that the same libraries were working with 4.2.0...

**Reply by szaimen** [reply]
Sorry, no idea.

**Reply by qudiqudi** [reply]
This is it then? Nextcloud AIO will miss a major feature? Both face recognition apps on the app-library are in-compatible with AIO.
Are there any plans on fixing this?

**Reply by szaimen** [reply]
The recognize is not maintained by me. If it has incompatibilities with Alpine Linux, it needs to be fixed there and not in AIO.
